pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -437,6 +454,54 @@ func buildGatewayListenerTLSContext(server *networking.Server, enableSds bool) *
 				},
 			}
 		}
+	} else if server.Tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
+		// global SDS must be enabled if the sds path is non-default, configure TLS with SDS
+		// TODO would prefer something less implicit here
+		if sdsConfig.path != "" {
+			// configure egress with SDS
+			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
+				CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
+					DefaultValidationContext: &auth.CertificateValidationContext{VerifySubjectAltName: server.Tls.SubjectAltNames},
+					ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(
+						authn_model.SDSRootResourceName, sdsConfig.path, sdsConfig.enableSdsTokenMount, sdsConfig.sdsUseK8sJwt, sdsConfig.metadata),
+				},
+			}
+			tls.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{
+				authn_model.ConstructSdsSecretConfig(
+					authn_model.SDSDefaultResourceName, sdsConfig.path, sdsConfig.enableSdsTokenMount, sdsConfig.sdsUseK8sJwt, sdsConfig.metadata)}
+		} else {
+			// global SDS disabled, fall back on using mounted certificates
```

Comment:
/cc @JimmyCYJ  @myidpt @costinm for comments.

We can start from global mesh config setting, but I learned we're working out some plan to make SDS enable gradually for workload. I think same things apply for Gateway.

IIUC, in another words, we need to check the pilot-agent passed params about `SDSEanbled` attributes as well. If that's true, but global.enabling is false, gateway sds enabling should still be true, to be consistent.

https://github.com/istio/istio/blob/master/pilot/pkg/kube/inject/inject.go#L220